src/Nox.Types/Types/Yaml/Yaml.cs

```diff
@@ -0,0 +1,254 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Nox.Types;
+
+/// <summary>
+///  Represents a Nox <see cref="Yaml"/> type and value object.
+/// </summary>
+public class Yaml : ValueObject<string, Yaml>
+{
+    private readonly List<YamlNode> _yamlNodes = new();
+    private bool _isValidated;
+
+    /// <summary>
+    /// Gets the equality components used for value comparison.
+    /// </summary>
+    /// <returns>The equality components.</returns>
+    protected override IEnumerable<KeyValuePair<string, object>> GetEqualityComponents()
+    {
+        var properties = GenerateKeyValuePairs();
```

Comment:
you can return this method directly no need to iterate

Reply:
Method was updated to return generated properties directly.